test/contacts.js

```diff
@@ -17,56 +38,83 @@ describe('contacts', function() {
+    it('should pass through a count', function() {
       const count = 10
+
       return hubspot.contacts.get({ count: 10 }).then(data => {
         expect(data).to.be.a('object')
         expect(data.contacts).to.be.a('array')
-        expect(data.contacts).to.have.length(count)
+        expect(data.contacts.length).to.be.at.most(10)
       })
     })
   })
 
   describe('getRecentlyModified', function() {
-    it('should return last 100 updated contacts', function() {
+    const modifiedContactsGetEndpoint = {
+      path: '/contacts/v1/lists/recently_updated/contacts/recent',
+      response: { contacts: [{}] },
+    }
+    fakeHubspotApi.setupServer({ getEndpoints: [modifiedContactsGetEndpoint] })
+
+    it('should return a list of contacts', function() {
       return hubspot.contacts.getRecentlyModified().then(data => {
         expect(data.contacts).to.be.an('array')
       })
     })
   })
 
   describe('getRecentlyCreated', function() {
-    it('should return last 100 updated contacts', function() {
+    const createdContactsGetEndpoint = {
+      path: '/contacts/v1/lists/all/contacts/recent',
+      response: { contacts: [{}] },
+    }
+    fakeHubspotApi.setupServer({ getEndpoints: [createdContactsGetEndpoint] })
+
+    it('should return a list of contacts', function() {
       return hubspot.contacts.getRecentlyCreated().then(data => {
         expect(data.contacts).to.be.an('array')
       })
     })
   })
 
   describe('getById', function() {
-    let contactId
+    let contactId = 123
+    const getContactByIdEndpoint = {
+      path: `/contacts/v1/contact/vid/${contactId}/profile`,
+      response: { vid: contactId },
+    }
+    fakeHubspotApi.setupServer({ getEndpoints: [getContactByIdEndpoint] })
 
     before(function() {
-      return hubspot.contacts.get().then(data => {
-        contactId = data.contacts[0].vid
-      })
+      if (process.env.NOCK_OFF) {
+        return hubspot.contacts.get().then(data => {
+          contactId = data.contacts[0].vid
+        })
+      }
     })
 
     it('should return a contact based on its id', function() {
       return hubspot.contacts.getById(contactId).then(data => {
         expect(data.vid).to.equal(contactId)
-        expect(data.properties.company).to.be.an('object')
       })
     })
   })
 
   describe('getByIdBatch', function() {
-    let contactIds
+    let contactIds = [123, 234, 345]
+    const getContactByIdsEndpoint = {
+      path: '/contacts/v1/contact/vids/batch',
+      response: { '123': {}, '234': {}, '345': {} },
+      query: { vid: contactIds },
+    }
+    fakeHubspotApi.setupServer({ getEndpoints: [getContactByIdsEndpoint] })
 
     before(function() {
-      return hubspot.contacts.get({ count: 10 }).then(data => {
-        contactIds = _.map(data.contacts, 'vid')
-      })
+      if (process.env.NOCK_OFF) {
+        return hubspot.contacts.get({ count: 3 }).then(data => {
+          contactIds = _.map(data.contacts, 'vid')
```

Comment:
Similar question here, will this fail if there are only 2 contacts?

Reply:
I think this would be fine with only 2 contacts since we map over the result.